### ISSUE TITLE
Saving machine causes a crash

### ISSUE THREAD

**Issue by MaddTheSane** [dirkwhoffmann/vAmiga#833]
If you go to Machine -> Save, vAmiga crashes.
I seems to crash here: https://github.com/dirkwhoffmann/vAmiga/blob/2b8e4c1899fed0c6a4c47c080daf2e9a8f3f3da1/Emulator/Base/CoreComponent.cpp#L144

The only thing I can think that might be wrong is that the `<<` operator might not be overridden.

**Comment by dirkwhoffmann** [comment]
Oops, the most fundamental functionality is broken in 2.6 🙄. Thanks for reporting!

The culprit is the objC wrapper (which will hopefully be obsolete one day). It juggles with void pointers, and I have introduced several new wrappers with the new SW architecture (coming from VirtualC64).

To get things right, I need to cast to `AmigaAPI` in this function (instead of `Amiga`):

```objc
+ (instancetype)makeWithAmiga:(AmigaProxy *)proxy
{
    AmigaAPI *amiga = (AmigaAPI *)proxy->obj;

    amiga->suspend();
    Snapshot *snapshot = new Snapshot(*(amiga->amiga));
    amiga->resume();
    
    return [self make:snapshot];
}
```

However, there is another bug. The emulator resets when loading a snapshot with `Open...`. I need to check if the same happens in VirtualC64. 

Personally, I always use drag-and-drop instead of `Open...` from the menu, which is why I didn't recognize this bug.

**Comment by dirkwhoffmann** [comment]
I tried the same a couple of weeks ago with similar error messages. Curiously, it works for VirtualC64. Thus, my plan is to clean up all APIs first (as part of porting VirtualC64's new emulator architecture to vAmiga) and then try to enable C++ interoperability again. 

BTW, I only utilize C++ interoperability in VirtualC64 to include native C++ header files at the moment, which allowed me to eliminate most `#ifdef` statements that check for C++. However, I am still using the ObjC proxy, because calling C++ functions directly did not work out for me as expected. The whole compile process slowed down massively. I've got the impression that Swift/C++ interoperability is in an early beta state. Thus, I'll wait for it to mature a bit before I try to elminate the ObjC proxy for good.
